processor/geoipprocessor/geoip_processor.go

```diff
@@ -115,3 +115,14 @@ func (g *geoIPProcessor) processAttributes(ctx context.Context, metadata pcommon
 
 	return nil
 }
+
+func (g *geoIPProcessor) shutdown(ctx context.Context) error {
+	for _, geoProvider := range g.providers {
+		err := geoProvider.Close(ctx)
+		if err != nil {
+			return err
```

Comment:
We should probably continue to close all providers instead of returning after finding the first provider that returns an error on close?

I know this processor currently only has one provider :upside_down_face: but if another provider is added, this code will be incorrect.

Reply:
Good point, wrapping error instead https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39379/commits/b97f2f6090a31a5ffd35f3c0f6d6e0137d5105cb

let me know if that works for you

Reply:
Perfect! :ok_hand: